test/uk/gov/hmrc/tai/connectors/HttpHandlerSpec.scala

```diff
@@ -46,286 +59,281 @@ class HttpHandlerSpec extends BaseSpec with BeforeAndAfterEach {
+      val ex = the[Upstream4xxResponse] thrownBy Await.result(responseFuture, 5 seconds)
+      ex.message must include("unknown response")
 
-        when(http.GET[HttpResponse](any())(any(), any(), any())).thenReturn(Future.successful(unauthorisedResponse))
-        val responseFuture = sut.getFromApiV2("")
-        val ex = the[HttpException] thrownBy Await.result(responseFuture, 5 seconds)
-        ex.message mustBe "\"unauthorised response\""
-      }
     }
+
   }
 
-  "putToApi" should {
+  "putToApi" must {
+
     "return OK" in {
-      when(http.PUT[DateRequest, HttpResponse](any(), any(), any())(any(), any(), any(), any()))
-        .thenReturn(Future.successful(HttpResponse(OK)))
 
-      val result = Await.result(sut.putToApi[DateRequest]("", DateRequest(LocalDate.now())), 5.seconds)
+      server.stubFor(
+        put(anyUrl())
+          .willReturn(aResponse().withStatus(OK)))
+
+      val result = Await.result(httpHandler.putToApi[DateRequest](testUrl, DateRequest(LocalDate.now())), 5.seconds)
 
       result.status mustBe OK
 
     }
 
-    "return Not Found exception" in {
-      when(http.PUT[DateRequest, HttpResponse](any(), any(), any())(any(), any(), any(), any()))
-        .thenReturn(Future.successful(HttpResponse(NOT_FOUND)))
+    "should return a NotFoundException when NOT_FOUND response" in {
+
+      server.stubFor(
+        put(anyUrl())
+          .willReturn(aResponse().withStatus(NOT_FOUND)))
 
       val result = the[NotFoundException] thrownBy Await
-        .result(sut.putToApi[DateRequest]("", DateRequest(LocalDate.now())), 5.seconds)
+        .result(httpHandler.putToApi[DateRequest](testUrl, DateRequest(LocalDate.now())), 5.seconds)
 
       result.responseCode mustBe NOT_FOUND
+
     }
 
-    "return Internal Server exception" in {
-      when(http.PUT[DateRequest, HttpResponse](any(), any(), any())(any(), any(), any(), any()))
-        .thenReturn(Future.successful(HttpResponse(INTERNAL_SERVER_ERROR)))
+    "should return a InternalServerException when INTERNAL_SERVER_ERROR response" in {
+
+      server.stubFor(
+        put(anyUrl())
+          .willReturn(aResponse().withStatus(INTERNAL_SERVER_ERROR).withBody("internal server exception")))
 
-      val result = the[InternalServerException] thrownBy Await
-        .result(sut.putToApi[DateRequest]("", DateRequest(LocalDate.now())), 5.seconds)
+      val ex = the[Upstream5xxResponse] thrownBy Await
+        .result(httpHandler.putToApi[DateRequest](testUrl, DateRequest(LocalDate.now())), 5.seconds)
+
+      ex.message must include("internal server exception")
 
-      result.responseCode mustBe INTERNAL_SERVER_ERROR
     }
 
-    "return Bad Request exception" in {
-      when(http.PUT[DateRequest, HttpResponse](any(), any(), any())(any(), any(), any(), any()))
-        .thenReturn(Future.successful(HttpResponse(BAD_REQUEST)))
+    "should return a BadRequestException when BAD_REQUEST response" in {
```

Comment:
This code is not throwing an exception, it is returning a HTTP status code

Reply:
Changing name to should return a BAD_REQUEST response code when BAD_REQUEST response